sound/soc/sof/ipc4-pcm.c

```diff
@@ -106,6 +285,17 @@ static int sof_ipc4_trigger_pipelines(struct snd_soc_component *component,
 		pipeline->state = state;
 	}
 
+	if (state == SOF_IPC4_PIPE_RUNNING) {
+		spcm->stream[substream->stream].stream_start_wclk =
+			SOF_IPC4_INVALID_STREAM_POSITION;
+		/*
+		 * wait for bus controller, codec and audio dsp are all active then driver can
+		 * get correct dma counter to init timecounter
+		 */
+		schedule_delayed_work(&spcm->stream[substream->stream].wclk_calibration,
+				      msecs_to_jiffies(SOF_IPC4_WCLK_CALIBRATION_DELAY_MS));
```

Comment:
does this work for capture streams?

Reply:
we can't get pipeline reg for capture.  Compared to playback, There is no mixing usage for capture and the input is started from dai, so stream_start_offset is zero and the llp counter should be accurate.

Reply:
@RanderWang what about the multi-stream support on capture. One Dai, multiple host pipelines. The same problem applies.

Reply:
Need to refine code:  remove pipeline reg finding for caputure and set stream_start_offset = 0 to calculate.